client/my-sites/themes/theme-showcase.jsx

```diff
@@ -191,7 +197,7 @@ class ThemeShowcase extends React.Component {
 	onTierSelect = ( { value: tier } ) => {
 		// In this state: tabFilter = [ ##Recommended## | All(1) ]   tier = [ All(2) | Free | Premium ]
 		// Clicking "Free" or "Premium" forces tabFilter from "Recommended" to "All"
-		if ( tier !== '' && tier !== 'all' && this.state.tabFilter === 'recommended' ) {
+		if ( tier !== '' && tier !== 'all' && this.state.tabFilter.key === 'recommended' ) {
 			this.setState( { tabFilter: 'all' } );
```

Comment:
* Go to theme showcase
* Switch to recommended tab
* Click on the 'free' button on the right side
* Expected to see: all themes tab is selected. Actually see: No tab is longer selected

```suggestion
			this.setState( { tabFilter: this.tabFilters.ALL } );
```